mage_integrations/mage_integrations/sources/paystack/streams/base.py

```diff
@@ -0,0 +1,166 @@
+                record,
+                self.catalog.schema.to_dict(),
+                metadata
+            )
+
+    def get_stream_data(self, data):
+        return [self.transform_record(item) for item in data]
+
+    def load_data(self, bookmarks: Dict = None, bookmark_properties: List = None, to_date: str = None):
+        table = self.TABLE
+        api_method = self.API_METHOD
+        done = False
+        sync_interval_in_mins = 2
+        bookmark_date = None
+
+        # Attempt to get the bookmark date from the state file (if one exists and is supplied).
+        LOGGER.info('Attempting to get the most recent bookmark_date for entity {}.'.format(self.ENTITY))
+        if bookmarks and bookmark_properties:
+            bookmark_date = bookmarks.get(bookmark_properties[0])
+
+        # If there is no bookmark date, fall back to using the start date from the config file.
+        if bookmark_date is None:
+            LOGGER.info('Could not locate bookmark_date from STATE file. Falling back to start_date from config.json instead.')
+            bookmark_datetime = parse(self.config.get('start_date'))
+            bookmark_date = bookmark_datetime.strftime('%Y-%m-%dT%H:%M:%SZ')
+
+        if to_date is None:
+            to_datetime = datetime.now(pytz.utc) - timedelta(minutes=sync_interval_in_mins)
+            to_date = to_datetime.strftime('%Y-%m-%dT%H:%M:%SZ')
+        sync_window = str([bookmark_date, to_date])
+        LOGGER.info("Sync Window {} for schema {}".format(sync_window, table))
+
+        params = {'from': bookmark_date, 'to': to_date}
+
+        LOGGER.info("Querying {} starting at {}".format(table, bookmark_date))
+
+        while not done:
+            max_date = to_date
+
+            response = self.client.make_request(
+                url=self.get_url(),
+                method=api_method,
+                params=params
+            )
+
+            meta = response.get('meta', dict())
+
+            if 'api_error_code' in response.keys():
+                if response['api_error_code'] == 'configuration_incompatible':
+                    LOGGER.error('{} is not configured'.format(response['error_code']))
+                    break
+
+            records = response.get('data')
+
+            # Get records from API response and transform
+            to_write = self.get_stream_data(records)
+
+            with singer.metrics.record_counter(endpoint=table) as ctr:
+                yield to_write
+
+                ctr.increment(amount=len(to_write))
+
+            if meta.get('page') >= meta.get('pageCount'):
+                LOGGER.info("Final page reached. Ending sync.")
+                done = True
+            else:
+                LOGGER.info("Advancing by one page.")
+                params['page'] = meta.get('page') + 1
+                bookmark_date = max_date
+
+    def get_class_path(self):
+        return os.path.dirname(inspect.getfile(self.__class__))
+
+    def load_schema_by_name(self, name):
```

Comment:
If the `load_schemas_from_folder` in `paystack/__init__.py` is not needed, this method, `get_schema` and `get_class_path` are not needed as well.